### ISSUE TITLE
About some confusion

### ISSUE THREAD

**Issue by ZhipengWu1996** [lizhuo-1994/mdp_testing#1]
Dear author, hello! Thank you for your inspiring work! I am primarily studying the application of your code in the MARL_CoopNavi scenario. However, I have found that the diffusion model seems to struggle in learning how to generate data that satisfies a specific distribution. Despite extensive training with a large amount of data, the diffusion model consistently generates data points outside the desired distribution range. I noticed that in the diffusion code, you enforce the generated data range to be limited between [-1, 1]. However, this restriction results in test cases always being at the boundary values, leading to poor performance in defect detection rates.

In summary, I seem to have difficulties achieving convergence when training the diffusion model. Could you please advise if there are any specific aspects to be mindful of in terms of parameter settings or other areas? Thank you very much!

**Comment by lizhuo-1994** [comment]
Thanks for supporting. I will run the code again and reproduce the problem you reported.
